magento/pwa-studio
# Better cypress results logging

## Description

When we added parallelization of cypress tests one of the cons was inability to see all the test results as a single unit at the end. This was because when a user decides to parallelize cypress tests, there will be `n` number of cypress threads and they wont be talking to each other to aggregate results at the end.

With this PR, we have added a new feature that will aggregate all the results and present them at the end in a readable tabular format. This functionality adds features to not only report the statues of all the tests but also aggregates the errors if any and prints them in red to catch the user's eye.

In case of all passes:

![image](https://user-images.githubusercontent.com/35203638/145625061-3a25165a-e27d-4ce9-899c-66e367b16843.png)

If something fails:

![image](https://user-images.githubusercontent.com/35203638/145625074-3f05e14f-572e-44da-b4c0-c933c644c190.png)

## Related Issue

Closes PWA-2453

### Verification Stakeholders

@dpatil-magento 

## Verification Steps

Run cypress tests as is and you should see results at the end.

`yarn test:headless -t 4 --url https://develop.pwa-venia.com/`
